### PR TITLE
Verify Nexus handler exists before distributing event to mirrors (SYN-7251)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/lib/nexus.py [hunk 1]
```diff
@@ -379,6 +379,13 @@ async def setindex(self, indx):
     async def _eat(self, item, indx=None):
 
         if self.donexslog:
+            nexsiden, event = item[:2]
```
Comment (vEpiphyte) [inline synapse/lib/nexus.py:382]: to reduce the amount of overhead while the nexus lock is held, could these be checked in `eat()` before we obtain the lock and fire the task?

Comment (Cisphyx) [reply synapse/lib/nexus.py]: Good point, yeah that's a better place for it. Moved it up there.

File: synapse/lib/nexus.py [hunk 2]
```diff
@@ -341,6 +341,12 @@ async def eat(self, nexsiden, event, args, kwargs, meta):
         if meta is None:
             meta = {}
 
+        if (nexus := self._nexskids.get(nexsiden)) is None:
+            raise s_exc.NoSuchName(mesg=f'No Nexus Pusher with iden {nexsiden}.', name=nexsiden)
```
Comment (invisig0th) [inline synapse/lib/nexus.py:345]: maybe `NoSuchIden` would be a tad more correct here?

Comment (Cisphyx) [reply synapse/lib/nexus.py]: yeah, updated

**Review by MichaelSquires — APPROVED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**